src/main/java/software/amazon/awssdk/crt/CRT.java

```diff
@@ -246,112 +245,65 @@ private static List<String> runProcess(String[] cmdArray) throws IOException {
-            if (platform != null){
-                String platformLibResourcePath = platform.getResourcePath(getCRuntime(os), libraryName);
-                if (platformLibResourcePath != null){
-                    libResourcePath = platformLibResourcePath;
+            tmpdirFile = new File(path).getAbsoluteFile();
+            tmpdirPath = tmpdirFile.getAbsolutePath();
+            if (tmpdirFile.exists()) {
+                if (!tmpdirFile.isDirectory()) {
+                    throw new IOException("not a directory: " + tmpdirPath);
                 }
+            } else {
+                tmpdirFile.mkdirs();
             }
 
-            try (InputStream in = CRT.class.getResourceAsStream(libResourcePath)) {
-                if (in == null) {
-                    throw new IOException("Unable to open library in jar for AWS CRT: " + libResourcePath);
-                }
-
-                // Copy from jar stream to temp file
-                try (FileOutputStream out = new FileOutputStream(tempSharedLib)) {
-                    int read;
-                    byte [] bytes = new byte[1024];
-                    while ((read = in.read(bytes)) != -1){
-                        out.write(bytes, 0, read);
-                    }
-                }
+            if (!tmpdirFile.canRead() || !tmpdirFile.canWrite()) {
+                throw new IOException("access denied: " + tmpdirPath);
             }
+        } catch (IOException ex) {
+            CrtRuntimeException rex = new CrtRuntimeException("Invalid directory: " + path);
+            rex.initCause(ex);
+            throw rex;
+        }
 
-            if (!tempSharedLib.setWritable(false)) {
-                throw new CrtRuntimeException("Unable to make shared library read-only");
-            }
+        String libraryName = System.mapLibraryName(CRT_LIB_NAME);
 
-            // load the shared lib from the temp path
-            System.load(tempSharedLib.getAbsolutePath());
-        } catch (CrtRuntimeException crtex) {
-            System.err.println("Unable to initialize AWS CRT: " + crtex);
-            crtex.printStackTrace();
-            throw crtex;
-        } catch (UnknownPlatformException upe) {
-            System.err.println("Unable to determine platform for AWS CRT: " + upe);
-            upe.printStackTrace();
-            CrtRuntimeException rex = new CrtRuntimeException("Unable to determine platform for AWS CRT");
-            rex.initCause(upe);
-            throw rex;
-        } catch (Exception ex) {
-            System.err.println("Unable to unpack AWS CRT lib: " + ex);
+        // Prefix the lib we'll extract to disk
+        String tempSharedLibPrefix = "AWSCRT_";
+        File tempSharedLib = null;
+        try{
+            tempSharedLib = File.createTempFile(tempSharedLibPrefix, libraryName, tmpdirFile);
+        }
+        catch (IOException ex){
+            System.err.println("Unable to create temp file to extract AWS CRT library" + ex);
```

Comment:
trivial: should the string end with ": " or something?
right now, `ex` is just going to get appended right after the letter "y"